File: client/my-community/src/api/community.ts
import service from '@/utils/net'

export const newCommunity = (data: string, jwt: string) => {
    return service.request({ url: '/community/new', method: 'post', data, headers: { 'Authorization': "Bearer " + jwt } })
}

export const searchCommunity = (data: FormData) => {
    return service.request({ url: '/community/search', method: 'post', data })
}

export const infoCommunity = (data: FormData, jwt: string) => {
    return service.request({ url: '/community/info', method: 'post', data, headers: { 'Authorization': "Bearer " + jwt } })
}

export const addCommunity = (data: string, jwt: string) => {
    return service.request({ url: '/community/add', method: 'post', data, headers: { 'Authorization': "Bearer " + jwt } })
}

export const editCommunity = (data: string, jwt: string) => {
    return service.request({ url: '/community/edit', method: 'post', data, headers: { 'Authorization': "Bearer " + jwt } })
}

export const addNote = (data: string, jwt: string) => {
    return service.request({ url: '/community/note/add', method: 'post', data, headers: { 'Authorization': "Bearer " + jwt } })
}

export const infoNote = (data: FormData) => {
    return service.request({ url: '/community/note/info', method: 'post', data })
}

File: app/lib/api/community.dart
import 'package:dio/dio.dart';
import 'package:my_community/utils/net.dart';

Future<Response> newCommunity(String data, jwt) {
  return Service().getAuthService(jwt).post('/community/new', data: data);
}

Future<Response> searchCommunity(FormData data) {
  return Service().getService().post('/community/search', data: data);
}

Future<Response> infoCommunity(FormData data, jwt) {
  return Service().getAuthService(jwt).post('/community/new', data: data);
}

Future<Response> addCommunity(String data, jwt) {
  return Service().getAuthService(jwt).post('/community/add', data: data);
}

Future<Response> editCommunity(String data, jwt) {
  return Service().getAuthService(jwt).post('/community/edit', data: data);
}

Future<Response> addNote(String data, jwt) {
  return Service().getAuthService(jwt).post('/community/edit', data: data);
}

Future<Response> infoNote(FormData data) {
  return Service().getService().post('/community/edit', data: data);
}